File: snakewatergun.py
# SNAKE WATER GUN PROJECT
# Snake vs. Water: Snake drinks the water hence wins.
# Water vs. Gun: The gun will drown in water, hence a point for water
# Gun vs. Snake: Gun will kill the snake and win.
# In situations where both players choose the same object, the result will be a draw
import random


def swg(comp, mine):
    if comp == "s" and mine == "g":
        return True
    elif comp == "w" and mine == "s":
        return True
    elif comp == "g" and mine == "w":
        return True
    else:
        return False


choice = ["s", "w", "g"]
comp = random.randint(0, 2)
comp = choice[comp]
mine = input("Enter 's' for snake or 'w' for water or 'g' for gun:")
win = swg(comp, mine)
if win:
    print("You Won")
else:
    print("You Lost")
print("The Computer Chose ", comp, " and you chose ", mine)

File: snakewatergun.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>
int snakewatergun(int you, int comp)
{
    if (you == comp)
    {
        return 0;
    }
    else if ((you == 's' && comp == 'w') || (you == 'w' && comp == 'g') || (you == 'g' && comp == 's'))
    {
        return 1;
    }
    else
    {
        return -1;
    }
}
int main()
{
    char you, comp;
    srand(time(0));
    int num = rand() % 100 + 1;
    if (num < 33)
    {
        comp = 's';
    }
    else if (num < 66)
    {
        comp = 'w';
    }
    else
    {
        comp = 'g';
    }
    printf("Type To Choose : S for snake,W for water,G for gun\n ");
    scanf("%c", &you);
    int result = snakewatergun(you, comp);
    if (result == 0)
    {
        printf("The Game Is Drawn\n");
    }
    else if (result == 1)
    {
        printf("You Win\n");
    }
    else
    {
        printf("You Lose\n");
    }
    printf("You Chose %c and Your Opponent Chose %c\n", you, comp);
}